### PR TITLE
add controllers

### PR DESCRIPTION
**Pull request by islomN**
added controllers (without actions), change gitignore

### CODE REVIEW

**Review by kirillta — CHANGES_REQUESTED** [review]
Waiting for changes

**Comment by islomN** [comment]
add only test controller action, this action for front check token

File: TipCatDotNet.Api/Controllers/CompanyController.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    [RequiredScope(ScopeRequiredByApi)]
+    public class CompanyController: BaseController
+    {
+        [HttpGet("{id}")]
+        public async Task<IActionResult> Get(int id)
+        {
+            return Ok();
+        }
+
+        [HttpGet("get-all")]
```
[the remark this answers is not in the document]
Comment (kirillta) [reply TipCatDotNet.Api/Controllers/CompanyController.cs:20]: By the way, Get All should be in another controller, so here we need only one Get, which returns only available companies for a token

File: TipCatDotNet.Api/Controllers/EmployeeController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Identity.Web.Resource;
+using TipCatDotNet.Api.Models.HospitalityFacilities;
+using TipCatDotNet.Api.Models.HospitalityFacilities.Enums;
+
+namespace TipCatDotNet.Api.Controllers
+{
+    [Authorize]
+    [Route("api/employee")]
```
Comment (kirillta) [inline TipCatDotNet.Api/Controllers/EmployeeController.cs:12]: `api/employees`

File: TipCatDotNet.Api/Data/Models/PermissionLevel.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace TipCatDotNet.Api.Data.Models
+{
+    public enum PermissionLevel: byte
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/PermissionLevel.cs:3]: Why do we need this, if we have [HospitalityFacilityPermissions](https://github.com/tipcat-net/tipcat-net-api/pull/5/files?file-filters%5B%5D=.cs&file-filters%5B%5D=.csproj&file-filters%5B%5D=.json#diff-b6321859ad6051ba014798c822d6e7ba1a083f7e49bf7fbfe668d81765841ab1) already?

File: TipCatDotNet.Api/Data/Models/User.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public class User
+    {
+        [Column("id")]
+        public int Id { get; set; }
+        
+        [Column("email")]
+        [StringLength(200)]
+        public string Email { get; set; }
+        
+        [Column("password_hash")]
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/User.cs:17]: We nave no data on passwords. Fields may be removed

File: TipCatDotNet.Api/Data/Models/User.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace TipCatDotNet.Api.Data.Models
+{
+    [Table("users")]
+    public class User
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/User.cs:8]: Why can't we merge both owners and employees to the one `Employee` table? There aren't much differences

File: TipCatDotNet.Api/Data/Models/Owner.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+
+namespace TipCatDotNet.Api.Data.Models
+{
+    [Table("accounts")]
+    public class Owner
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/Owner.cs:8]: Use one term for program classes and database entities. This is much safer and clearer.

File: TipCatDotNet.Api/Data/Models/ModelStatus.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+namespace TipCatDotNet.Api.Data.Models
+{
+    public enum ModelStatus: byte
+    {
+        InActive = 0,
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/ModelStatus.cs:5]: Inactive is one word

**Review by kirillta — CHANGES_REQUESTED** [review]

File: TipCatDotNet.Api/Data/Models/Employee.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        [Column("created")]
+        public DateTime Created { get; set; }
+                
+        [Column("modified")]
+        public DateTime Modified { get; set; }
+        
+        [Column("company_id")]
+        public int CompanyId { get; set; }
+        
+        [ForeignKey("CompanyId")]
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/Employee.cs:46]: Here should be `AccountId`, I presume

File: TipCatDotNet.Api/Data/Models/Employee.cs [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    public class Employee
+    {
+        [Column("id")]
+        public int Id { get; set; }
+        
+        [Column("user_id")]
+        public int UserId { get; set; }
+        
+        [ForeignKey("UserId")]
+        public virtual User User { get; set; }
```
Comment (kirillta) [inline TipCatDotNet.Api/Data/Models/Employee.cs:17]: Please, move keys to the bottom of the class

**Review by kirillta — CHANGES_REQUESTED** [review]

File: TipCatDotNet.Api/TipCatDotNet.Api.csproj [hunk 10]
```diff
@@ -1,15 +1,21 @@
...
     <TargetFramework>net5.0</TargetFramework>
     <Nullable>enable</Nullable>
     <LangVersion>9</LangVersion>
     <AssemblyName>TipCatDotNet.Api</AssemblyName>
     <RootNamespace>TipCatDotNet.Api</RootNamespace>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
+    <NoWarn>$(NoWarn);1591</NoWarn>
+    <UserSecretsId>f9ca6b90-5ae3-495a-8935-d9b80b74806d</UserSecretsId>
+  </PropertyGroup>
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
```
Comment (kirillta) [inline TipCatDotNet.Api/TipCatDotNet.Api.csproj:12]: Debug only condition looks like an error

**Review by kirillta — APPROVED** [review]

**Review by kirillta — CHANGES_REQUESTED** [review]
Changes required

**Review by evm-andrey — APPROVED** [review]

**Comment by kirillta** [comment]
These features merged in another branch

**Pull request closed without merging**